cmd/metal-api/internal/datastore/event.go

```diff
@@ -35,3 +38,41 @@ func (rs *RethinkStore) CreateProvisioningEventContainer(ec *metal.ProvisioningE
 func (rs *RethinkStore) UpsertProvisioningEventContainer(ec *metal.ProvisioningEventContainer) error {
 	return rs.upsertEntity(rs.eventTable(), ec)
 }
+func (rs *RethinkStore) ProvisioningEventForMachine(machineID, event, message string) (*metal.ProvisioningEventContainer, error) {
```

Comment:
I think that it would be better if we factor this code out into a separate event package in order to make it accessible from grpc and rest. This will also give the future-planned FSM a place to live + it keeps the datastore dumb, which we tried to achieve everywhere else.

Reply:
unsure, only event related logic here